### ISSUE TITLE
Question on permissions

### ISSUE THREAD

**Issue by IzzySoft** [tharunbirla/FetchIt#12]
Your app is [listed at IzzyOnDroid](https://apt.izzysoft.de/packages/com.tharunbirla.fetchit), where the scanner reports:

```
! repo/com.tharunbirla.fetchit_4.apk declares sensitive permission(s):
  android.permission.READ_EXTERNAL_STORAGE android.permission.MANAGE_EXTERNAL_STORAGE
  android.permission.READ_MEDIA_IMAGES android.permission.READ_MEDIA_VIDEO
  android.permission.READ_MEDIA_AUDIO
```

Could you please clarify what those permissions are needed for? Your Readme speaks of "storing the downloads" – but at least the last 3 permissions listed above should not be needed for that. Thanks in advance!

Oh, and btw: be welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) and link to your app at IzzyOnDroid e.g. from your Readme :wink:

**Comment by tharunbirla** [comment]
Hi @IzzySoft , thanks for listing the app.

**WRITE_EXTERNAL_STORAGE**: For android 10 and below
**MANAGE_EXTERNAL_STORAGE** : For android 11 and above
(It may not be needed for standard storage as it give broad access to storage)
**READ_MEDIA_VIDEO**: For android 13 and above.

The other permissions (`MANAGE_EXTERNAL_STORAGE`, `READ_MEDIA_IMAGES` and `READ_MEDIA_AUDIO`) are excessive. I will streamline the permissions and retain only `READ_MEDIA_VIDEO` in the future release.

Thank you!

**Comment by RoestVrijStaal** [comment]
As long as I am able to save stuff at any storage and path, it might be okay to remove. 

Permission purists often give usability of the app a lower priority.

**Comment by IzzySoft** [comment]
Well, which one is for Android ≤ 10 and which for ≥ 11 is clear. My question was rather their purpose. Going by the app description, FetchIt is a video downloader, not e.g. a video player. So basically, one directory (and the ability to create/use sub-directories) should be sufficient.

> Permission purists often rank usability of the app to a lower priority.

The point here is transparency. Sure, you should not request permissions your app doesn't need. But with the partly crazy permission system of Android (especially storage permissions between Android 5 and 10 are hard to grasp even for those otherwise quite familiar with how Android works) people might wonder or even "suspect" – so at IzzyOnDroid, we prefer to have at least sensible permissions explained, with a short note. Example from another app:

![image](https://github.com/user-attachments/assets/d829d721-c8c0-47b8-8571-2de94b44e405)

(explanations could be kept pretty short there as the logic was clear from the app purpose/description)

> MANAGE_EXTERNAL_STORAGE : For android 11 and above
> The other permissions (MANAGE_EXTERNAL_STORAGE, READ_MEDIA_IMAGES and READ_MEDIA_AUDIO) are excessive.

Small overlap :wink: First thanks for cleaning up those not needed! Can we have a short note for those remaining then? Like:

```yaml
android.permission.READ_EXTERNAL_STORAGE: ??
android.permission.WRITE_EXTERNAL_STORAGE: used to save videos to any storage location on Android ≤ 10
android.permission.MANAGE_EXTERNAL_STORAGE: used to save videos to any storage location on Android ≥ 11
android.permission.READ_MEDIA_VIDEO: ??
```

Do you really need access to "any storage", so SAF without MANAGE is not sufficient? Just asking for clarification, as pointed out. I already admitted, storage permissions with or without SAF are a mess at least between Android 5 (your app's minimum requirement) and 10…

**Comment by RoestVrijStaal** [comment]
> So basically, one directory (and the ability to create/use sub-directories) should be sufficient.
And this is what I am afraid for and hate. No ability to save stuff to other places than **Downloads** or **Videos**. More tedious to use, because I have to use a filemanager app to move files to correct places. Annoying.

**Comment by tharunbirla** [comment]
> > So basically, one directory (and the ability to create/use sub-directories) should be sufficient.
> 
> And this is what I am afraid for and hate.
> 
> No ability to save stuff to other places than **Downloads** or **Videos** on the device's internal storage. More tedious to use, because I have to use a filemanager app to move files to correct places. Annoying. User unfriendly.

Hi @RoestVrijStaal ,

I understand your concern about limiting the save locations (I did it due to security reasons).

I'll try different way to offer more granular control over save locations, possibly user-selectable options. This way, users can choose where to save videos.

I'll keep you updated on any changes and improvements.

**Comment by tharunbirla** [comment]
> IIRC, SAF lets you pick almost any directory (with a few exceptions). But well, I'm neither forcing nor pushing anyone there, I'm just suggesting. SAF gives the person using an app much more granular choice about what that app can access, so if you have it installed you can be rather sure which places it can access and which not. While MANAGE means it can access the full storage. 🤷‍♂️
> 
> TL;DR: if it's no option, I'll accept that. We'd then just need to replace the "question marks" in above list 😉
> 
> Oh, and of course it's for @tharunbirla to decide. Who I'm sure will consider the thoughts of the people using the app (here: you), too 😉

Hi @IzzySoft ,

Apologies for the late reply.

I understand the concerns regarding the excessive permissions. I'll definitely streamline them in the next update. I'll remove all unnecessary permissions and retain only the essential ones for downloading videos.

I appreciate your (@IzzySoft and @RoestVrijStaal ) feedback and suggestions. I'll work on improving the app's permission usage and user experience 😊

**Comment by IzzySoft** [comment]
Thanks @tharunbirla! Is there something I shall meanwhile put as explanation? At IzzyOnDroid, we try to provide as much transparency as we can, so next to "sensitive" permissions we put explanations what they are needed/used for (see above). For FetchIt, that would be the four outlined above. The best I can come up with by guessing based on the background is:

```yaml
android.permission.READ_EXTERNAL_STORAGE: ??
android.permission.WRITE_EXTERNAL_STORAGE: used to save videos to any storage location on Android ≤ 10
android.permission.MANAGE_EXTERNAL_STORAGE: used to save videos to any storage location on Android ≥ 11
android.permission.READ_MEDIA_VIDEO: ??
```

`READ_EXTERNAL_STORAGE` currently is requested explicitly (it would still be granted if not requested explicitly, as then it would be granted implicitly due to `WRITE_EXTERNAL_STORAGE'). So for the two with question marks I don't know what to put. The other two I've just added to the app's "green list" as shown in this comment.

**Comment by tharunbirla** [comment]
**Permission review:**

1.  `android.permission.READ_EXTERNAL_STORAGE`
      - **Required**: No (Implicitly granted with `WRITE_EXTERNAL_STORAGE` on Android ≤ 10, will remove the explicit request, as it's not necessary for the app's functionality)
      
2.  `android.permission.READ_MEDIA_VIDEO`
      - **Required**: No (Not utilized, will remove)

The forthcoming updates will reflect the adjustments outlined above. I will ensure that FetchIt only requests the necessary permissions. Thank you @IzzySoft  for your understanding and continued support! :smiley:

**Comment by IzzySoft** [comment]
Thanks! Will add the explanation for `READ_EXTERNAL_STORAGE` then, leaving the other one "untouched". The latter will be shown in warning color then, but gone once you've removed it. Already live now:

![image](https://github.com/user-attachments/assets/11584521-18db-4616-bbec-58a7df89e6b1)

Leaving this issue open then for you (reminder for removal of `READ_MEDIA_*`), feel free to close if you no longer need it open.

**Comment by tharunbirla** [comment]
Hey @IzzySoft I removed the extra permissions please verify 😊 d481175ff5dbcf96fe5edea87a00c50df14bcfef

**Comment by IzzySoft** [comment]
Yay, success:

![image](https://github.com/user-attachments/assets/c9dcdff2-8bc9-4552-88db-f2055a5a4c96)

Shield's not yet up as the RB run happened an hour before the release. Let me give that a manual kick… Done, shield should be up now, too.

Thanks a lot!!!
